### PR TITLE
Api provides an endpoint for showing a single recipe

### PR DESCRIPTION
**Pull request by elvitak**
### This PR contains:

- add ingredient model with unit spec and factory,
- migrate to add ingredients to recipe
- add associations between recipe and ingredients
- add show route
- add show method in recipe controller
- add error handler for show action

https://www.pivotaltracker.com/story/show/181136364

@Mljungho @DefyCab @giacoletti

### CODE REVIEW

File: app/models/recipe.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class Recipe < ApplicationRecord
   validates_presence_of :title, :instructions
```
Comment (Mljungho) [inline app/models/recipe.rb:2]: We should validate the presence of :ingredients as well.

**Review by Mljungho — CHANGES_REQUESTED** [review]
I only had one comment. 

@giacoletti how about you?

File: app/models/recipe.rb (hunk 1 above)
Comment (elvitak) [reply app/models/recipe.rb:2]: we validate ingredients (name, amount, units) in Ingredient model

Comment (Mljungho) [reply app/models/recipe.rb:2]: Yes, but we sold not be able to create a recipe with "ingredient => Nil"

**Pull request merged**